test/e2e/autoscaling_utils.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"strconv"
+	"sync"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/util"
+
+	. "github.com/onsi/ginkgo"
+)
+
+const (
+	seconds     = 10
+	sleepTime   = time.Duration(10) * time.Second
+	requestSize = 100
+)
+
+// ConsumingRC is a tool for testing. It helps create specified usage of CPU or memory (Warnig: memory not supported)
+// typical use case:
+// rc.ConsumeCPU(600)
+// // ... check your assumption here
+// rc.ConsumeCPU(300)
+// // ... check your assumption here
+type ConsumingRC struct {
+	name          string
+	milicores     int
+	mutex         *sync.Mutex
+	framework     *Framework
+	stopConsuming bool
+}
+
+// creates new ConsumingRC
+func NewConsumingRC(name string, replicas int, framework *Framework) *ConsumingRC {
+	startService(framework.Client, framework.Namespace.Name, name, replicas)
+	rc := &ConsumingRC{
+		name:          name,
+		milicores:     0,
+		mutex:         &sync.Mutex{},
+		framework:     framework,
+		stopConsuming: false,
+	}
+	go rc.consumeCPU()
+	return rc
+}
+
+// consumes given number of CPU
+func (rc *ConsumingRC) ConsumeCPU(milicores int) {
+	rc.mutex.Lock()
+	rc.milicores = milicores
+	rc.mutex.Unlock()
+}
+
+func (rc *ConsumingRC) consumeCPU() {
+	defer GinkgoRecover()
+	for !rc.stopConsuming {
+		count := rc.milicores / requestSize
+		rest := rc.milicores - count*requestSize
+		rc.sendConsumeCPUrequests(count, requestSize, seconds)
+		if rest != 0 {
+			rc.sendOneConsumeCPUrequest(rest, seconds)
+		}
+		time.Sleep(sleepTime)
+	}
```

Comment:
I would:
- ask pods to consume resources much longer (>1 min).
- update them frequently so that your use case with autoscaling is satisfied.
- calculate the sleep time based on the amount of time the communication used. 
- consider sending requests on multiple goroutines/threads.